src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -628,6 +639,34 @@ private async Task EnqueueWorkItemAfterDiffAsync(Solution oldSolution, Solution
                 var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
 
                 await EnqueueWorkItemAsync(oldProject.GetRequiredDocument(documentId), newProject.GetRequiredDocument(documentId)).ConfigureAwait(continueOnCapturedContext: false);
+
+                var oldProjectSourceGeneratedDocuments = await oldProject.GetSourceGeneratedDocumentsAsync(CancellationToken.None).ConfigureAwait(false);
+                var oldProjectSourceGeneratedDocumentsById = oldProjectSourceGeneratedDocuments.ToDictionary(static document => document.Id);
+                var newProjectSourceGeneratedDocuments = await newProject.GetSourceGeneratedDocumentsAsync(CancellationToken.None).ConfigureAwait(false);
+                var newProjectSourceGeneratedDocumentsById = newProjectSourceGeneratedDocuments.ToDictionary(static document => document.Id);
```

Comment:
❗ This is a non-cancellable call to run source generators on an IDE fast path

Reply:
➡️ The `_shutdownToken` is now passed through. Resolved by gating this on `WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace`